src/Collection/CollectionInterface.php

```diff
@@ -775,13 +775,13 @@ public function insert(string $path, $values): CollectionInterface;
     /**
      * Returns an array representation of the results
      *
-     * @param bool $preserveKeys whether to use the keys returned by this
+     * @param bool $keepKeys Whether to use the keys returned by this
      * collection as the array keys. Keep in mind that it is valid for iterators
      * to return the same key for different elements, setting this value to false
      * can help getting all items if keys are not important in the result.
      * @return array
      */
-    public function toArray(bool $preserveKeys = true): array;
+    public function toArray(bool $keepKeys = true): array;
```

Comment:
I'm don't think we include named parameter names as part of our backwards compatibility promises, so this should be ok.

Reply:
I cleaned up a lot before 4.2, but I don't know if we want to make a guarantee. I'd hate to change it too much. Maybe we have a policy in cake 5 since it's php 8+.